h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -42,27 +44,127 @@
+        this.count -= c;
+        return c * this.share;
+      }
+
+      int consumeAll() {
+        return consume(Integer.MAX_VALUE);
+      }
+    }
+
+    private boolean canAllocate = true;
+    private final LinkedList<Work> allocations = new LinkedList<>();
+
+    WorkAllocations allocate(Algo algo, int count, JobType type, int workShare) {
+      if (!canAllocate) throw new IllegalStateException("can't allocate new work");
+
+      allocations.add(new Work(algo, count, type, workShare));
+      return this;
+    }
+
+    void end() {
+      canAllocate = false;
+    }
+
+    void remove(Algo algo) {
+      final Iterator<Work> iter = allocations.iterator();
+      while (iter.hasNext()) {
+        if (algo.equals(iter.next().algo)) iter.remove();
+      }
+    }
+
+    Work getAllocation(Algo algo, JobType workType) {
+      for (Work alloc : allocations) {
+        if (alloc.algo == algo && alloc.type == workType) return alloc;
+      }
+      return null;
+    }
+
+    private int sum(List<Work> workItems) {
+      int tot = 0;
+      for (Work item : workItems) {
+        tot += (item.count * item.share);
+      }
+      return tot;
+    }
+
+    int remainingWork() {
+      return sum(allocations);
+    }
+
+    int remainingWork(JobType type) {
+      List<Work> selected = new ArrayList<>(allocations.size());
+      for (Work item : allocations) {
+        if (item.type == type) selected.add(item);
+      }
+      return sum(selected);
+    }
+
+  }
+
   private final static boolean verifyImmutability = true; // check that trainingFrame hasn't been messed with
   private final static SimpleDateFormat fullTimestampFormat = new SimpleDateFormat("yyyy.MM.dd HH:mm:ss.S");
   private final static SimpleDateFormat timestampFormatForKeys = new SimpleDateFormat("yyyyMMdd_HHmmss");
 
-  // TODO: UGH: this should be dynamic, and it's easy to make it so
-  //   just turn this enum into a class...
+  // if we need to make the Algo list dynamic, we should just turn this enum into a class...
   // NOTE: make sure that this is in sync with the exclude option in AutoMLBuildSpecV99
-  public enum algo {
+  public enum Algo implements algo {
     GLM,
     DRF,
     GBM,
     DeepLearning,
     StackedEnsemble,
-//    XGBoost,
+    XGBoost,
     ;
 
     String urlName() {
       return this.name().toLowerCase();
     }
   }
 
+  /**
+   * for external use only.
+   * Keeping this mapping for customers consuming the Java API directly
+   * @deprecated use {@link Algo} instead
+   */
+  @Deprecated
+  public interface algo {
+    @Deprecated public static final Algo GLM = Algo.GLM;
+    @Deprecated public static final Algo DRF = Algo.DRF;
+    @Deprecated public static final Algo GBM = Algo.GBM;
+    @Deprecated public static final Algo DeepLearning = Algo.DeepLearning;
+    @Deprecated public static final Algo StackedEnsemble = Algo.StackedEnsemble;
+    @Deprecated public static final Algo XGBoost = Algo.XGBoost;
+  }
```

Comment:
@michalkurka : I had many conflicts and compilation issues (`algo` being used as a class and as a variable in many places then) when rebasing the renaming changes on this branch.
I suggest the above deprecation to be used for external legacy use only, besides deprecation will inform the user so that we will be able to remove it safely at some point.
WDYT?